### DISCUSSION TITLE
TOTP Codes Seem To Fall Out Of Sync

### DISCUSSION THREAD

**Discussion by bonedaddy** [dani-garcia/vaultwarden#3054] [Q&A]
### Subject of the issue
TOTP verification codes sometimes seem to fall out of sync from which the server they were issued from. Starting from version 1.24 (earliest I can recall this happening) every so I will be completely unable to use TOTP codes copied from the bitwarden extension, or the vaultwarden webvault. Any attempt in using them will result in the TOTP code being rejected.

I believe the issue is due to there being some synchronization issue between vaultwarden, and the source that issued the TOTP codes. My reasoning is that I can copy a TOTP code, attempt to ssh to one of my VPS, paste the TOTP code and have it rejected twice, but then on the third usage attempt have the TOTP code be accepted.

This issue only appears to affect TOTP codes issued by `google-authenticator` cli command.

**Comment by BlackDex** [comment]
TOTP codes are generate on the clients. These clients use the time provide by there host system, either mobile or computer etc.

The only thing vaultwarden does is provide the encrypted data.

**Reply by BlackDex** [reply]
So, check the time on the device you use to generate these codes, it's probably out if sync with the actual time.

**Reply by stefan0xC** [reply]
Tip: You can easily check the time (and how much your device's clock is off) with sites like https://www.time.gov/ and https://uhr.ptb.de/

**Reply by bonedaddy** [reply]
Yes this was correct, misconfiguration with chronyd server. thanks!
